idl/src/build.rs

```diff
@@ -95,6 +95,10 @@ fn build(program_path: &Path, resolution: bool, no_docs: bool) -> Result<Idl> {
             "ANCHOR_IDL_BUILD_RESOLUTION",
             if resolution { "TRUE" } else { "FALSE" },
         )
+        .env(
+            "ANCHOR_IDL_BUILD_PROGRAM_PATH",
+            program_path.to_str().unwrap(),
+        )
```

Comment:
We should be able to just pass `program_path` here since [`std::process::Command.env`](https://doc.rust-lang.org/std/process/struct.Command.html#method.env) accepts `AsRef<OsStr>`:

```suggestion
        .env("ANCHOR_IDL_BUILD_PROGRAM_PATH", program_path)
```

Reply:
done